src/DynamoCore/Engine/ShortestQualifiedNameReplacer.cs

```diff
@@ -69,12 +71,68 @@ node.datatype.UID < (int)PrimitiveType.MaxPrimitive)
             return node;
         }
 
+        /// <summary>
+        /// If node is a DesignScript.BuiltIn.Get.ValueAtIndex(exp1, exp2) function call node.
+        /// it is transformed into a node with indexing operator e.g. exp1[exp2].
+        /// </summary>
+        /// <param name="node">input node</param>
+        /// <param name="indexExp">output node</param>
+        /// <returns>true if input expression is of the form DesignScript.BuiltIn.Get.ValueAtIndex(exp1, exp2), false otherwise</returns>
+        private bool TryBuildIndexingExpression(IdentifierListNode node, out ArrayNameNode indexExp)
+        {
+            indexExp = null;
+            if(node.RightNode is FunctionCallNode fcn)
+            {
+                if(fcn.Function.Name == ProtoCore.AST.Node.BuiltinValueAtIndexMethodName)
+                {
+                    if(node.LeftNode.ToString() == ProtoCore.AST.Node.BuiltinGetValueAtIndexTypeName)
+                    {
+                        var exp1 = fcn.FormalArguments[0];
+                        var exp2 = fcn.FormalArguments[1];
+
+                        if (exp1 is ArrayNameNode ann)
+                        {
+                            if(exp1 is IdentifierListNode iln1)
+                            {
+                                if (!TryBuildIndexingExpression(iln1, out indexExp))
+                                {
+                                    indexExp = ann.CreateInstance();
+                                }
+                            }
+                            else indexExp = ann.CreateInstance();
+
+                            if (exp2 is IdentifierListNode iln2)
+                            {
+                                if (TryBuildIndexingExpression(iln2, out ArrayNameNode index))
+                                {
+                                    if (indexExp.ArrayDimensions != null)
+                                    {
+                                        indexExp = new IdentifierNode(indexExp.ToString());
+                                    }
+                                    indexExp.ArrayDimensions = new ArrayNode(index, null);
+                                    return true;
+                                }
+                            }
+                            if(indexExp.ArrayDimensions != null)
+                            {
+                                indexExp = new IdentifierNode(indexExp.ToString());
+                            }
+                            indexExp.ArrayDimensions = new ArrayNode(exp2, null);
+                            return true;
+                        }
+                    }
+                }
+            }
+            return false;
+        }
 
         public override AssociativeNode VisitIdentifierListNode(IdentifierListNode node)
         {
             if (node == null)
                 return null;
 
+            if(TryBuildIndexingExpression(node, out ArrayNameNode indexExp)) return indexExp;
```

Comment:
I'm having a hard time remembering exactly how these visitors work - but if you return the new index expression here, does that mean it won't also have class name shortening applied? For example something like:

`DesignScript.BuiltIn.Get.ValueAtIndex(Package.Package.MyCoolMethodThatReturnsAList(),1)`

Reply:
Ah, you're right, good catch. It should not return from the visit function immediately but continue on to try name shortening. Will make the change.